### PR TITLE
i18n updates

### PR DESCRIPTION
**Pull request by SelmanKahya**

### CODE REVIEW

File: src/i18n/i18n-server.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        ...options,
+        backend: {
+          loadPath: `${appSrc}/i18n/translations/{{lng}}.json`,
+        },
+        /*
+        detection: {
+          // order and from where user language should be detected
+          order: ['path', 'session', 'querystring', 'cookie', 'header'],
+
+          /* CONSIDER USING THESE OPTIONS:
```
Comment (SelmanKahya) [inline src/i18n/i18n-server.js]: left these for now, it could be a starting point for optimizations.

File: src/i18n/index.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// eslint-disable-next-line import/no-mutable-exports
+let i18n;
+
+if (process.env.NODE_ENV === 'test') {
```
Comment (SelmanKahya) [inline src/i18n/index.js]: i18n.t is a mock function in test env.

File: src/server.js [hunk 3]
```diff
@@ -31,82 +22,73 @@ if (process.env.RAZZLE_ASSETS_MANIFEST) {
 const publicDir = process.env.RAZZLE_PUBLIC_DIR || 'public';
 
 const server = express();
-i18n
```
Comment (SelmanKahya) [inline src/server.js]: @trkaplan  this block clutters server.js, lets keep things simple

File: src/server.js [hunk 4]
```diff
@@ -31,82 +22,73 @@ if (process.env.RAZZLE_ASSETS_MANIFEST) {
...
-  .use(Backend)
-  .use(i18nextMiddleware.LanguageDetector)
-  .init(
-    {
-      backend: {
-        addPath: `${appSrc}/locales/{{lng}}/{{ns}}.missing.json`,
```
Comment (SelmanKahya) [inline src/server.js]: not needed

File: src/server.js [hunk 5]
```diff
@@ -31,82 +22,73 @@ if (process.env.RAZZLE_ASSETS_MANIFEST) {
...
-        loadPath: `${appSrc}/locales/{{lng}}/{{ns}}.json`,
-      },
-      preload: ['en'],
-    },
-    () => {
```
Comment (SelmanKahya) [inline src/server.js]: ideally, we should wait till i18n is initialized, it is fine for now. server will start while i18n is loading.

File: src/server.js [hunk 6]
```diff
@@ -31,82 +22,73 @@ if (process.env.RAZZLE_ASSETS_MANIFEST) {
...
-        .use('/locales', express.static(`${appSrc}/locales`))
-        .use(express.static(publicDir))
-        .get('/*', (req, res) => {
-          // STORE
-          const preloadedState = {};
-          const history = createMemoryHistory();
-          const store = setupStore(history, preloadedState);
-          const context = {};
-          const markup = renderToString(
-            <I18nextProvider i18n={req.i18n}>
```
Comment (SelmanKahya) [inline src/server.js]: @trkaplan there is no need for i18next provider. it simply makes things much harder. you need to wrap your component wherever you need translations. i find that unnecessary.

File: src/server.js [hunk 7]
```diff
@@ -31,82 +22,73 @@ if (process.env.RAZZLE_ASSETS_MANIFEST) {
...
-            </I18nextProvider>,
-          );
 
-          // Grab the initial state from our Redux store
-          const finalState = store.getState();
-          const cssAssetUrl = _get(assets, 'css/fonts.css');
-          const jsAssetUrl = _get(assets, 'client.js');
+if (process.env.NODE_ENV !== 'test') {
+  // not using i18n middleware on test env
+  server.use(i18nextMiddleware.handle(i18n));
```
Comment (SelmanKahya) [inline src/server.js]: @trkaplan this is why your test case was failing

**Comment by SelmanKahya** [comment]
dev is broken, test case if failing. I am going to merge this, let me know if you have any comments.

File: src/i18n/index.js (hunk 2 above)
Comment (trkaplan) [reply src/i18n/index.js]: 👍

File: src/server.js (hunk 3 above)
Comment (trkaplan) [reply src/server.js]: 👍

**Pull request merged**